src/server.rs

```diff
@@ -406,32 +438,61 @@ impl<F, H> Server<F, H> {
         self
     }
 
+    /// Run the server in testing mode.
+    ///
+    /// This will disable exposing metrics and health.
+    pub fn testing(mut self, testing: bool) -> Self {
```

Comment:
I agree. I'll think of a better name. I wish rust allowed public `#[cfg(test)]` API, but you can't 😞

Reply:
I've renamed the method to `disable_health_and_metrics` since thats what it did. Figured its fine to keep it specific since we haven't needed more specific things internally as of yet.